src/controllers/user.controller/fetchAllUsers/index.js

```diff
@@ -24,6 +24,12 @@ const getAllUsers = catchAsync(async (req, res) => {
     (blockedUser) => blockedUser.user_id
   );
 
+  const blockedUserIds = new Set(
+    usersToExcludeResponse.result.data.user_blocked_users.map(
+      (blockedUser) => blockedUser.blocked_user_id
+    )
+  );
+
   const response = await Hasura(getAllUsersQuery, { cognito_sub, usersToExclude });
```

Comment:
You are already excluding users who are blocked with the usersToExlude variables.

So all the returned users will always be non-blocked users...